README.md

```diff
@@ -24,95 +22,135 @@ Well, it's our lucky day!
 
-This is a _super_ powerful idea that makes it much easier for us to build programs that can do _many_ things. Instead of having to operate on each element using a `for` loop, we can instead very succinctly pass a function to `Array.prototype.forEach` and let the function do the work for us!
+| Pass # | `even` | `index` |       `array`       |
+|:------:|:------:|:-------:|:-------------------:|
+|   1    |   0    |    0    | [0, 2, 4, 6, 8, 10] |
+|   2    |   2    |    1    | [0, 2, 4, 6, 8, 10] |
+|   3    |   4    |    2    | [0, 2, 4, 6, 8, 10] |
+|   4    |   6    |    3    | [0, 2, 4, 6, 8, 10] |
+|   5    |   8    |    4    | [0, 2, 4, 6, 8, 10] |
+|   6    |   10   |    5    | [0, 2, 4, 6, 8, 10] |
 
-## Functions as Arguments
+It's possible to provide the same functionality with a regular `for` loop, but to do so we would have to jump through some hoops that `.forEach()` mercifully abstracts away for us. To replicate `.forEach()` with a `for` loop, we could define the function ahead of time and then invoke it inside of the loop...
 
-When a language allows functions to be used as arguments, we say that the language treats functions as **first-class objects**. Oftentimes, these functions serve as **callbacks**, which do just what their name implies: they are _called back_, potentially with arguments, and do their work on bits of data that weren't available outside of the function.
+```js
+var evens = [0, 2, 4, 6, 8, 10];
+
+function evenFn(even, index, array) {
+  console.log(`${even} is not odd!`);
+}
+
+for (let i = 0; i < evens.length; i++) {
+  evenFn(evens[i], i, evens);
+}
+```
 
-For example, let's say that we have a function for logging the square of a number called, appropriately enough, `square`.
+...but that seems like too many extra lines for such a simple operation. We could also use an IIFE, an immediately-invoked function expression, to make our code a bit more compact, but it's starting to get far more confusing than the elegant simplicity of a `.forEach()`:
 
-``` javascript
+```js
+var evens = [0, 2, 4, 6, 8, 10];
+
+for (let i = 0; i < evens.length; i++) {
+  (function (even, index, array) {
+  debugger;
+    console.log(`${even} is not odd!`);
+  })(evens[i], i, evens);
+}
+```
+
+`.forEach()` provides a wonderfully compact, readable syntax for invoking a function on every member of an array. It's a powerful tool that makes it easier for us to perform complex array operations in just a few lines of code.
+
+## Functions as Arguments
+Functions in JavaScript are **first-class objects**, and one of the best perks of that special status is that we can pass functions as arguments to other functions. Oftentimes, these passed-in functions serve as **callbacks**, which do just what the name implies: they are _called back_, potentially with arguments, and do their work on data that weren't available at runtime.
```

Comment:
Also I think we go into callbacks more later. Tell the students that because it's another rabbit hole they'll fall down